PyTorch/Classification/ConvNets/image_classification/dali_autoaugment.py

```diff
@@ -0,0 +1,325 @@
+                          fill_value=128,
+                          inverse_map=False)
+
+
+@op_magnitude_range(0, 250, auto_symmetric=True)
+def translateY(samples, parameter):
+    mt = fn.transforms.translation(offset=[0, parameter])
+    return fn.warp_affine(samples,
+                          matrix=mt,
+                          fill_value=128,
+                          inverse_map=False)
+
+
+@op_magnitude_range(0, 30, auto_symmetric=True)
+def rotate(samples, parameter):
+    return fn.rotate(samples, angle=parameter, fill_value=128)
+
+
+@op_magnitude_range(0.1, 1.9)
+def brightness(samples, parameter):
+    return fn.brightness(samples, brightness=parameter)
+
+
+@op_magnitude_range(0.1, 1.9)
+def contrast(samples, parameter):
+    return fn.contrast(samples, contrast=parameter)
+
+
+@op_magnitude_range(0.1, 1.9)
+def color(samples, parameter):
+    return fn.saturation(samples, saturation=parameter)
+
+
+@op_magnitude_range(0, 4)
+def posterize(samples, parameter):
+    nbits = np.round(parameter).astype(np.int32)
+    mask = np.array(255 ^ (2**(8 - nbits) - 1), dtype=np.uint8)
+    return samples & mask
+
+
+@op_magnitude_range(0, 110)
+def solarize(samples, threshold):
+    samples_inv = 255 - samples
+    mask_left = samples < np.uint8(threshold)
+    mask_right = types.Constant(1) - mask_left
+    return fn.cast(mask_left * samples + mask_right * samples_inv,
+                   dtype=types.UINT8)
+    # return samples
+
+
+@op_magnitude_range(0, 256)
+def solarize_add(samples, shift):
+    samples_shifted = fn.cast_like(samples + shift, samples)
+    mask_left = samples < 128
+    mask_right = types.Constant(1) - mask_left
+    return fn.cast_like(mask_left * samples_shifted + mask_right * samples,
+                        samples)
+    # return samples
+
+
+@op_magnitude_range(0.1, 1.9)
+def sharpness(samples, magnitude):
+    blur = np.array([[1, 1, 1], [1, 5, 1], [1, 1, 1]], dtype=np.float32) / 13
+    ident = np.array([[0, 0, 0], [0, 1, 0], [0, 0, 0]], dtype=np.float32)
+    kernel = (1 - magnitude) * blur + magnitude * ident
+    return fn.experimental.filter(samples, kernel)
+
+
+@op_magnitude_range(0, 1)
+def invert(samples, _):
+    return fn.cast_like(255 - samples, samples)
+
+
+# TODO(klecki): No-op
+@op_magnitude_range(0, 1)
+def equalize(samples, _):
+    # return fn.experimental.equalize(samples)
+    return samples
+
+
+@op_magnitude_range(0, 1)
+def autocontrast(samples, _):
+    lo, hi = fn.reductions.min(samples,
+                               axes=[-3, -2]), fn.reductions.max(samples,
+                                                                 axes=[-3, -2])
+    lo = fn.expand_dims(lo, axes=[0, 1])
+    hi = fn.expand_dims(hi, axes=[0, 1])
+    return fn.cast_like((samples - lo) * (255 / (hi - lo)), samples)
+
+
+def subpolicy(op_desc0, op_desc1):
+    op0, p0, mag0 = op_desc0
+    op1, p1, mag1 = op_desc1
+
+    def subpolicy_impl(images, rand0, rand1, is_pos0, is_pos1):
+        if rand0 < p0:
```

Comment:
fyi: the `fn.random.uniform` default range is `[-1,1]`, so the ops were run with probability `0.5(1 + p)` instead of `p`.

Reply:
🤦‍♂️